### ISSUE TITLE
Blocking numbers

### ISSUE THREAD

**Issue by MatejKovacic** [GrapheneOS/os-issue-tracker#2428]
First of all, thanks for a really great OS you are providing.

I would like to have a feature to block calls (and SMS messages):
- from callers with hidden caller ID
- from callers not on my contact list
- blocking a specific number, BUT with a pattern (for instance blocking numbers from a specific country or area code)

The latter could be useful, because in my country some people are receiving a lot of scam and spam calls from one of the neighbouring countries. So blocking the whole country or a specific area code in that country would be useful feature.

What do you think?

**Comment by andre4ik3** [comment]
There's an app on F-Droid called Silence that does what you want. It's a bit old and broken, but it does the job. (Make sure you set it as the caller ID app in default apps).

If you only want the first two, you can use Simple Dialer app, also from F-Droid. In settings, there is an option to "block incoming calls from non-contacts" (under blocking). It's a lot simpler, but I found it works more reliably. (Again, set whichever one you want to filter calls as caller ID app)

For SMS filtering I use QKSMS. Maybe the Simple SMS app also has some option, I don't know, I haven't used it.

Point is, no need for this to be an operating-system function. There are already APIs that let you do this and several apps that implement them.

**Comment by thestinger** [comment]
Silence is missing important security patches.

**Comment by MatejKovacic** [comment]
Exactly. That is why these features for default valler app in Graphene would be preffered option.

**Comment by andre4ik3** [comment]
> Silence is missing important security patches.

Didn't know that! Well either way, as I said I use the option in Simple Dialer app now, and it does get updated every once in a while, so I think (hope) that that doesn't have any security issues. It's just an on-off toggle under the "call blocking" settings.

> Exactly. That is why these features for default valler app in Graphene would be preffered option.

The default dialer app is from AOSP and was deprecated a while back. As I understand GrapheneOS will eventually have a replacement for it but that won't happen for a while. When there is a replacement, maybe it will get added, but I am just saying you can use caller ID apps to block unknown calls in the meantime.

**Comment by roamer4223** [comment]
> Silence is missing important security patches.

Do you have any more detail about this? Like the type of patches and the severity. Or could you point me somewhere its been discussed in more detail? As I've looked through their issues page and can't find anything there. Or if you know an alternative app of course.

Only because I use this app and can't find a substitute that does all the things I need. I have been worried for a while as it hasn't been updated in ages...

Thank you

**Comment by grigio** [comment]
+1

**Comment by burdges** [comment]
It'd be nice if the blocking supported a phone number range somehow.

**Comment by chuchana** [comment]
Would you consider making the blocking of unknown numbers an option that could be toggled individually for each SIM?

**Comment by datafoo** [comment]
> Would you consider making the blocking of unknown numbers an option that could be toggled individually for each SIM?

I would very much like that as well. One of my SIM is from Free Mobile, a french operator which had a missive data leak some time ago. Tired of spam calls, I switched to a new phone number less than 20 days ago and only informed some contacts about it. I already received 3 automated spam calls but they did not seem to know anything about me. So I would be perfectly happy to "block numbers not in Contacts" on that particular SIM but I cannot do that globally for all SIMs.

**Comment by grigio** [comment]
> l. One of my SIM is from Free Mobile, a french operator which had a missive data leak some time ago. Tired of spam calls, I switched to a new phone number less than 20 days ago and

You can use SpamBlocker on f-droid

**Comment by chuchana** [comment]
> You can use SpamBlocker on f-droid

This one: https://f-droid.org/packages/spam.blocker/ ?

**Comment by kgaranger** [comment]
> > You can use SpamBlocker on f-droid
> 
> This one: https://f-droid.org/packages/spam.blocker/ ?

I confirm that SpamBlocker can be used to block calls on a specific SIM. See https://github.com/aj3423/SpamBlocker/issues/169#issuecomment-3617874545 for details.

**Comment by chuchana** [comment]
Thanks! I've also been using it successfully for a while now

> I confirm that SpamBlocker can be used to block calls on a specific SIM. See [aj3423/SpamBlocker#169 (comment)](https://github.com/aj3423/SpamBlocker/issues/169#issuecomment-3617874545) for details.
